docs/api.md

```diff
@@ -0,0 +1,339 @@
+```
+
+Update the development cookie when running locally:
+
+```ruby
+if Rails.env.development?
+  cookies["govuk_account_session"] = {
+    value: "<govuk_account_session value>",
+    domain: "dev.gov.uk",
+  }
+end
+```
+
+### End the user's session if an endpoint returns a `401: Unauthenticated`
+
+If an endpoint returns a `401: Unauthenticated` response, then the
+user must no longer be considered logged in.
+
+Set the `GOVUK-Account-End-Session` response header:
+
+```ruby
+response.headers["GOVUK-Account-End-Session"] = "1"
+```
+
+Expire the development cookie when running locally:
+
+```ruby
+if Rails.env.development?
+  cookies["govuk_account_session"] = {
+    value: "",
+    domain: "dev.gov.uk",
+    expires: 1.second.ago,
+  }
+end
+```
+
+### Ensure responses are not cached
+
+Any user-visible response which involves calling this API must return
+appropriate response headers to either forbid caching:
+
+```ruby
+response.headers["Cache-Control"] = "no-store"
+```
+
+Or specify that the response depends on the user:
+
+```ruby
+response.headers["Vary"] = [response.headers["Vary"], "GOVUK-Account-Session"].compact.join(", ")
+```
+
+
+## Example API usage
+
+This is an example of updating an attribute for the current user.
+
+The `update` method is the controller action.  The `disable_cache`,
+`fetch_session_identifier`, `set_session_identifier`, and `logout!`
+methods would be re-used by other actions which call this API.
+
+```ruby
+before_action :disable_cache, only: %i[update]
+before_action :fetch_session_identifier, only: %i[update]
+
+def update
+  head :unauthorised and return unless @session_identifier
+
+  api_response = GdsApi.account_api.set_attributes(
+    govuk_account_session: @session_identifier,
+    attributes: { example_attribute: params[:attribute_value] },
+  ).to_h
+
+  set_session_identifier api_response["govuk_account_session"]
+rescue GdsApi::HTTPUnauthorized
+  logout!
+  render plain: "you have been logged out"
+end
+
+private
+
+def disable_cache
+  response.headers["Cache-Control"] = "no-store"
+end
+
+def fetch_session_identifier
+  @session_identifier =
+    if request.headers["HTTP_GOVUK_ACCOUNT_SESSION"]
+      request.headers["HTTP_GOVUK_ACCOUNT_SESSION"]
+    elsif Rails.env.development?
+      cookies["govuk_account_session"]
+    end
+end
+
+def set_session_identifier(session_identifier)
+  return unless session_identifier
+
+  response.headers["GOVUK-Account-Session"] = @session_identifier
+
+  if Rails.env.development?
+    cookies["govuk_account_session"] = {
+      value: @session_identifier,
+      domain: "dev.gov.uk",
+    }
+  end
+end
+
+def logout!
+  @session_identifier = nil
+
+  response.headers["GOVUK-Account-End-Session"] = "1"
+
+  if Rails.env.development?
+    cookies["govuk_account_session"] = {
+      value: "",
+      domain: "dev.gov.uk",
+      expires: 1.second.ago,
+    }
+  end
+end
+```
+
+
+## API endpoints
```

Comment:
@barrucadu apologies I forgot to include some feedback on this section. Best practice for API reference documentation is to include the following:

- Endpoints and methods
- Parameters and arguments
- Example requests and responses for each endpoint with example parameters
- Error response codes

This content has the first two. Is it possible to include the example requests and responses, and error response codes? Happy to discuss at any point.

Reply:
:+1: I've added response codes and example requests & responses.